### PR TITLE
EXP System (Made by Volco and HH)

### PR DESCRIPTION
**Pull request by DeathlyPlays**
I fixed it up and made it nef so you can use it

### CODE REVIEW

**Comment by DeathlyPlays** [comment]
Lemme fix up this file ig for Travis

**Comment by ghost** [comment]
Do we need it on spacial?

File: chat.js [hunk 1]
```diff
@@ -211,8 +211,9 @@ class CommandContext {
 						this.room.add((this.room.type === 'chat' ? (this.room.type === 'chat' ? '|c:|' + (~~(Date.now() / 1000)) + '|' : '|c|') : '|c|') + this.user.getIdentity(this.room.id) + '|' + message);
 						this.room.messageCount++;
 					}
+					this.room.add(`|c|${this.user.getIdentity(this.room.id)}|${message}`).update();
```
Comment (HoeenCoder) [inline chat.js:214]: Im p sure this will screw up emotes / shadow ban

File: spacialgaze-plugins/exp.js [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+			if (level === 2 || level === 4) {
+				//reward = "-3 Stardust";
+				Db.currency.set(user.userid, Db.currency.get(user.userid, 0) + 2);
+				//user.sendTo(room, 'You have earned ' + reward + ' for level up!');
+			}
+			if (level === 6 || level === 7 || level === 9 || level === 11 || level === 14 || level === 16 || level === 15 || level === 16 || level === 18 || level === 19) {
+				//reward = "-5 Stardust";
+				Db.currency.set(user.userid, Db.currency.get(user.userid, 0) + 5);
+				//user.sendTo(room, 'You have earned ' + reward + ' for level up!');
+			}
```
Comment (HoeenCoder) [inline spacialgaze-plugins/exp.js:113]: This is messy, really messy. It should be cleaned up and optimized.

**Review by HoeenCoder — COMMENTED** [review]
Idk if EXP should be on SG, this needs to be run through policy. Also this is pretty much the same messy code as last time.

File: chat.js (hunk 1 above)
Comment (DeathlyPlays) [reply chat.js:214]: It might, hence why Exiled emotes don't work in PMs I guess?  Shadow ban works though

**Pull request closed without merging**